### DISCUSSION TITLE
Using 3d models

### DISCUSSION THREAD

**Discussion by Sheepyhead** [bevyengine/bevy#1225] [Q&A]
There needs to be a definitive answer on how to use 3D models in Bevy. Thus far half the glb files I manage to create only partially work and often only parts of the full mesh are loaded in correctly. I have no idea what the "#Mesh0/Primitive0" syntax actually does or how to modify a model to fit this syntax, even though it's apparently mandatory for loading 3d meshes. Ideally I would think you could simply give the file path and the file is loaded in full, but every time I've tried this, nothing loads. It's extremely unclear which file types are supported, and how to create and export 3D models and animations that work with Bevy.

**Comment by FrankenApps** [comment]
Are you using v. 0.4?
Most of the time this happened to me, because my shader expected certain information that was not encoded in the gltf vertecies. For example VertexNormals and VertexColors. So the fix would be to use a fitting shader.
